pkg/kubelet/kubeletconfig/controller.go

```diff
@@ -43,9 +43,16 @@ const (
 	configTrialDuration = 10 * time.Minute
 )
 
+type TransformFunc func(kc *kubeletconfig.KubeletConfiguration) error
```

Comment:
I'm not 100% sure if this is the pattern I want, because it looks like a generic utility but we have a very specific use-case: enforce flag precedence so the final config combination is validated before choosing a config.

At the very least, we should warn people in a comment, and maybe change the name to something less generic.

Reply:
Alternatively, we could make it something generic, and use it in the e2e tests. https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/util.go#L106 is essentially the same function.

Reply:
It's more about how it's used in the controller; it's an extension point that we only want to use as a last resort.

Reply:
Though I don't see how we get away from it as long as we have to enforce flag precedence.

Reply:
I think we'll go with this for now. I added a warning to the comment above the field on the controller and to the NewController constructor.